### DISCUSSION TITLE
can you please tell me which part of those code helps on reading CRF and other files?

### DISCUSSION THREAD

**Discussion by johnhhu2020** [DHI/Mike1D-SDK-examples#16] [Q&A] [answered]
hello gedaskir, nice to know you. do  you think your code 'ResultDataExtract.py' can read CRF, PRF, TRF, XRF files? i would be so appreciate, if you cound enlight me how to make it happened. Please help, you can email me for detailed discussions. cheers

**Comment by JesperGr** [comment]
Hello johnhhu

That should work out of the box, by just replacing the res1d input filename with the crf/prf filename when executing the script (unless you have a very old crf/prf file, which is not supported by MIKE 1D).

**Comment by johnhhu2020** [comment]
Hello JesperGr, thanks for your quick reply, whereas I still need your help on the using of this script. Say I want to query every information from a PRF file, what should be my sample code? Or if I want to save the FloodRate info into a output csv file. many thanks

**Comment by johnhhu2020** [comment]
I have tried 
ipy.exe ResultDataExtract.py MyFileName.PRF output.csv node:-:Node_1377     (it tells me 'WaterLevel' data is available), 
but when I code 
ipy.exe ResultDataExtract.py MyFileName.PRF output.csv node:WaterLevel:Node_1377     (not working)

What can I do to extract data from a PRF file properly?

**Comment by gedaskir** [comment]
Hello johnhhu,

As Jesper mentioned the script **ResultDataExtract.py** should work for CRF/PRF files out of the box. I just tested it with IronPython 3.4.0-alpha1 and installation of Mike Zero 2020 Update 1 and it does export from CRF to CSV file. As far as I can see you also can retrieve the available quantities, so it shows that Mike libraries are loaded on your machine. You mention that the export of water level does not work. What kind of errors do you get?

Additionally, improvements need to be made for the example script to be able to export all of the information from PRF file easily.

**Comment by johnhhu2020** [comment]
OK, I got the company computer. Here is the error code:

'''
C:\Program Files\IronPython 3.4>ipy.exe ResultDataExtract.py 10mmBase.PRF output.csv reach:WaterLevel:Link_99
Traceback (most recent call last):
File "ResultDataExtract.py", line 862, in
File "ResultDataExtract.py", line 858, in Main
File "ResultDataExtract.py", line 578, in Export
PermissionError: [Errno 13] Permission denied: 'output.csv'
'''

I got the MIKE Zero 2021 installed, and opened WIndows Command Prompt and cd to IronPython environment, put PRF file and XRF files inside, and ran this code
"ipy.exe ResultDataExtract.py 10mmBase.PRF output.csv reach:WaterLevel:Link_99"

The wield stuff have happened manytimes on my company computer.

**Comment by gedaskir** [comment, marked as answer]
The problem here is clear. You do not have sufficient permissions to write the CSV file in **C:\Program Files\IronPython 3.4**. Try to move the files (**ResultDataExtract.py**, **10mmBase.PRF**, **output.csv**) to some folder where you have write access and execute the wanted command
```console
C:\Program Files\IronPython 3.4\ipy.exe ResultDataExtract.py 10mmBase.PRF output.csv reach:WaterLevel:Link_99
```
from within that folder.

**Comment by johnhhu2020** [comment]
Thank you Gedaskir, It seems my company account does not support files writting under C disk, I ran the Command as Administrator and it works on PRF file, whereas when I was trying to extract data in another file - a TRF or XRF, here is what I got (below commands should show me what available quantities are available in those files): 

C:\Program Files\IronPython 3.4>ipy.exe ResultDataExtract.py 10mmBase.TRF Reach_TRF_Link_999_FlowVelocity_Output.csv reach:
Traceback (most recent call last):
  File "ResultDataExtract.py", line 862, in <module>
  File "ResultDataExtract.py", line 817, in Main
  File "ResultDataExtract.py", line 278, in __init__
  File "ResultDataExtract.py", line 292, in SetupFilter
SystemError: Object reference not set to an instance of an object.

C:\Program Files\IronPython 3.4>ipy.exe ResultDataExtract.py result-1A.XRF Reach_XRF_Link_999_FlowVelocity_Output.csv reach
Traceback (most recent call last):
  File "ResultDataExtract.py", line 862, in <module>
  File "ResultDataExtract.py", line 817, in Main
  File "ResultDataExtract.py", line 278, in __init__
  File "ResultDataExtract.py", line 292, in SetupFilter
SystemError: Object reference not set to an instance of an object.

C:\Program Files\IronPython 3.4>ipy.exe ResultDataExtract.py 10mm_BuildingBase.CRF Reach_XRF_Link_999_FlowVelocity_Output.csv reach
Available Reach names in file:
(nothing)

**Comment by gedaskir** [comment]
The first two errors point to some kind of Mike1D bug. Also the TRF/XRF may be of older format not supported by Mike1D or the files have no data. It is possible to say what is really happening only by investigating the files. It makes sense that there is no reach information for CRF file, because it is a Catchment Result File (CRF).

**Comment by johnhhu2020** [comment]
Thank you Gedaskir and JesperGr, your comments helped me a lot. I'll check those files and understand Mike product better.

**Comment by johnhhu2020** [comment]
And again, thanks for your time, patient and kindness. Sincerely

**Comment by gedaskir** [comment]
Nice to see that the script is used. Though some improvements are needed. Also glad to help :-)

**Comment by johnhhu2020** [comment]
problem solved
